src/app/components/GoalScorerDisplay.tsx

```diff
@@ -1,54 +1,62 @@
-import React, { SyntheticEvent, Dispatch } from 'react'
-import { TeamScore } from './TeamScorer' 
+import React, { SyntheticEvent, Dispatch } from "react";
+import { TeamScore } from "./TeamScorer";
 
 type Props = {
-    teamAScoreObject: TeamScore
-    setTeamAScoreObject: Dispatch<React.SetStateAction<TeamScore>>
-}
-
-
-function GoalScorerDisplay({teamAScoreObject, setTeamAScoreObject}: Props) {
-
-// Adds the name from input into the goal array under scorer 
-    function updateScorer(e: SyntheticEvent<HTMLFormElement>){
-        e.preventDefault()
-        const newArray : Array = teamAScoreObject.goal
-        newArray[e.target.id] =  {
-            scorer: e.target.name.value, 
-            period: newArray.period, 
-            time: newArray.time, 
-            id: newArray.id
-        }
-        setTeamAScoreObject({
-            goal: newArray,
-            behind: teamAScoreObject.behind
-
-        })
-
+  teamAScoreObject: TeamScore;
+  setTeamAScoreObject: Dispatch<React.SetStateAction<TeamScore>>;
+};
+
+function GoalScorerDisplay({ teamAScoreObject, setTeamAScoreObject }: Props) {
+  let inputRef = React.createRef<HTMLInputElement>();
```

Comment:
Not sure if you have used refs much but they are handy when you need to make references to DOM elements:https://react.dev/learn/referencing-values-with-refs